src/Lucene.Net/Util/HybridArray.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+namespace Lucene.Net.Util;
+
+public static class HybridArray
+{
+    // simple threshold: 64KB (guarantees ArrayPool won't exceed 85KB LOH limit)
+    private const int LOH_THRESHOLD = 64 * 1024;
+
+    public static unsafe IArray<T> Create<T>(int length, UnmanagedStringArray.Type type) where T : unmanaged
+    {
+        if (length < 0)
+            throw new ArgumentOutOfRangeException(nameof(length));
+
+        if (length * sizeof(T) > LOH_THRESHOLD)
```

Comment:
Why don't make _everything_ unmanaged?

Reply:
I think using the ArrayPool will be more cost-effective for small allocations. For long arrays, using this threshold will accommodate `8,192` elements, while for TermInfo (which is `32` bytes), it will contain up to `2,048` elements. Smaller segments with fewer cached terms are more likely to be merged frequently than larger segments, which creates more allocation and deallocation cycles.

Reply:
This is sth that we can check, but in general I think that calling AllocHGlobal and FreeHGlobal for each of those each time will have a much bigger penalty. Smaller allocations should be managed and come from a pool IMO in this case.